controllers/snapshotenvironmentbinding_controller.go

```diff
@@ -157,12 +157,13 @@ func addTargetIfNotExists(secret *rapi.RemoteSecret, target rapi.RemoteSecretTar
 }
 
 func removeTarget(secret *rapi.RemoteSecret, target rapi.RemoteSecretTarget) {
-	for idx := range secret.Spec.Targets {
-		existingTarget := secret.Spec.Targets[idx]
-		if targetsMatch(existingTarget, target) {
-			secret.Spec.Targets = append(secret.Spec.Targets[:idx], secret.Spec.Targets[idx+1:]...)
+	tmp := secret.Spec.Targets[:0]
+	for _, existingTarget := range secret.Spec.Targets {
+		if !targetsMatch(existingTarget, target) {
+			tmp = append(tmp, existingTarget)
 		}
 	}
+	secret.Spec.Targets = tmp
 }
 
 func targetsMatch(target1, target2 rapi.RemoteSecretTarget) bool {
```

Comment:
Btw I would argue that ClusterCredentialsSecret SHOULD NOT be part of the equality, because all it means is that the operator will use different credentials to create the same secret.

Reply:
so the proposal is to exlude `ClusterCredentialsSecret` from targets comparing ?